src/replicache.ts

```diff
@@ -39,6 +44,12 @@ const storageKeyName = (name: string) => `/replicache/root/${name}`;
 /** The maximum number of time to call out to getDataLayerAuth before giving up and throwing an error. */
 const MAX_REAUTH_TRIES = 8;
 
+type BroadcastData = {
+  root?: string;
+  changedKeys: ChangedKeysMap;
+  index?: string;
```

Comment:
It's not immediately apparent how this field is used. Suggest either add some comments and/or making it more self-documenting?

```
type DataChange = {
  changedKeys: ChangedKeysMap;
}

type IndexChange = {
  index: string;
}

type BroadcastData = {root?: string} & (DataChange | IndexChange};
```

Reply:
Added a comment. Splitting the types makes TS sad